src/includes/traits/Plugin/OptionUtils.php

```diff
@@ -10,9 +10,13 @@
      *
      * @since 151002 Improving multisite compat.
      *
+     * @param bool  $intersect Discard options not present in $this->default_options
+     *
      * @return array Plugin options.
+     *
+     * @note $intersect should be `false` when this method is called via a VS upgrade routine or during inital startup on when upgrading. See https://git.io/viGIK
      */
-    public function getOptions()
+    public function getOptions($intersect = true)
     {
         if (!($options = $this->options)) { // Not defined yet?
```

Comment:
@raamdev It might be helpful if you added another new parameter to this method that you might call `$refresh`, and if `$refresh` is true, this check for existing options would be bypassed, thereby pulling the options from the underlying `get_site_option()` call again.

With that in place, you should then be able to remove the double call to `getOptions()` in the `setup()` routine, and instead, in the `checkVersion()` method, if there _is_ a new version, before the VS upgrades run, make another call to refresh the options array, with `$intersect = false` and `$refresh = true`.